plancton/__init__.py

```diff
@@ -210,18 +211,19 @@ def _read_conf(self):
       self.conf["max_docks"] = 0
     if not isinstance(self.conf["docker_cmd"], list):
       self.conf["docker_cmd"] = self.conf["docker_cmd"].split(" ")
-    if self.conf["influxdb_url"] and not "#" in self.conf["influxdb_url"]:
-      # Invalid InfluxDB URL: disable monitoring
-      self.conf["influxdb_url"] = None
-    self.logctl.debug("Configuration:\n%s" % json.dumps(self.conf, indent=2))
+    if isinstance(self.conf["influxdb_url"], str):
+      self.conf["influxdb_url"] = set([self.conf["influxdb_url"]])
+    elif isinstance(self.conf["influxdb_url"], list):
+      self.conf["influxdb_url"] = set(filter(lambda x: "#" in x, self.conf["influxdb_url"]))
+    else:
+      self.conf["influxdb_url"] = set()
+    self.logctl.debug("Configuration:\n%s" % json.dumps(self.conf, indent=2, default=list))
 
   # Set up monitoring target.
   def _influxdb_setup(self):
-    if not self.conf["influxdb_url"]:
-      self.streamer = lambda *x,**y: True
-      return
-    baseurl,db = self.conf["influxdb_url"].split("#")
-    self.streamer = InfluxDBStreamer(baseurl=baseurl, database=db)
+    self.streamers = set([ x for x in self.streamers if x.baseurl+"#"+x.database in self.conf["influxdb_url"] ])
```

Comment:
This line still does not work. It always removes all streamers (checked now with "printf debug"). I am fixing it.